tests/test_sandbox/test_ecs_fargate.py

```
@@ -1396,3 +1379,259 @@ def test_split_keeps_unrelated_env_in_stable(self):
+        registry, repo, ref = ImageBuilder._parse_image_ref(f"reg.example.com/ns/img:1.2@{self._DIGEST}")
+        assert (registry, repo, ref) == ("reg.example.com", "ns/img", self._DIGEST)
+
+    def test_mirror_tag_handles_digest_pinned_source_refs(self):
+        src = f"acme/task@{self._DIGEST}"
+        tag = ImageBuilder.get_ecr_mirror_tag(src, self._DIGEST)
+        assert tag == ImageBuilder.get_ecr_mirror_tag(src, self._DIGEST)
+        assert tag != ImageBuilder.get_ecr_mirror_tag(src, "sha256:" + "cd" * 32)
+        assert len(tag) <= 128
+
+    @patch(f"{_PATCH_PREFIX}.ImageBuilder.image_exists_in_ecr", return_value=False)
+    def test_mirroring_respects_build_parallelism(self, _mock_exists):
```

Ruff (0.16.1)

[warning] Fixture `_mock_exists` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)